anime_tracker/src/models/animeSearch.ts

```diff
@@ -0,0 +1,149 @@
+export type OverviewArray = Overview[]
+export interface Overview {
+  mal_id: number
+  url: string
+  images: Images
+  trailer: Trailer
+  approved: boolean
+  titles: Title[]
+  title: string
+  title_english?: string
+  title_japanese: string
+  title_synonyms: string[]
+  type: string
+  source: string
+  episodes?: number
+  status: string
+  airing: boolean
+  aired: Aired
+  duration: string
+  rating: string
+  score?: number
+  scored_by?: number
+  rank?: number
+  popularity: number
+  members: number
+  favorites: number
+  synopsis?: string
+  background?: string
+  season?: string
+  year?: number
+  broadcast: Broadcast
+  producers: Producer[]
+  licensors: Licensor[]
+  studios: Studio[]
+  genres: Genre[]
+  explicit_genres: any[]
+  themes: Theme[]
+  demographics: Demographic[]
+}
+
+export interface Images {
+  jpg: Jpg
+  webp: Webp
+}
+
+export interface Jpg {
+  image_url: string
+  small_image_url: string
+  large_image_url: string
+}
+
+export interface Webp {
+  image_url: string
+  small_image_url: string
+  large_image_url: string
+}
+
+export interface Trailer {
+  youtube_id?: string
+  url?: string
+  embed_url?: string
+  images: Images2
+}
+
+export interface Images2 {
+  image_url?: string
+  small_image_url?: string
+  medium_image_url?: string
+  large_image_url?: string
+  maximum_image_url?: string
+}
+
+export interface Title {
+  type: string
+  title: string
+}
+
+export interface Aired {
+  from?: string
+  to?: string
+  prop: Prop
+  string: string
+}
+
+export interface Prop {
+  from: From
+  to: To
+}
+
+export interface From {
```

Comment:
what is _From_? what is _To_? and why are they the same types? code duplication (a lot in this file)

Reply:
[#](https://github.com/sane120/animeTracker/pull/11#discussion_r1348538688) Same as in here. Is there a better way to handle the JSON call?